client/ts/demo/main.ts

```diff
@@ -0,0 +1,562 @@
+import {
+  PublicKey,
+  Connection,
+  Keypair,
+  sendAndConfirmTransaction,
+  Transaction,
+  TransactionInstruction,
+  SystemProgram,
+  SendTransactionError,
+} from '@solana/web3.js';
+import { ManifestClient } from '../src/client';
+import { Market } from '../src/market';
+import {
+  createAssociatedTokenAccountIdempotent,
+  createMint,
+  getAssociatedTokenAddressSync,
+  mintTo,
+} from '@solana/spl-token';
+import { airdropSol, getClusterFromConnection } from '../src/utils/solana';
+import { OrderType, PROGRAM_ID } from '../src/manifest';
+import { FIXED_MANIFEST_HEADER_SIZE } from '../src/constants';
+
+class DummyTrader {
+  constructor(
+    public connection: Connection,
+    public keypair: Keypair,
+    public marketAddress: PublicKey,
+    public name: string,
+    private commands: string[],
+    private client: ManifestClient,
+  ) {}
+
+  public static async create(
+    connection: Connection,
+    keypair: Keypair,
+    marketAddress: PublicKey,
+    name: string,
+    commands: string[],
+  ) {
+    const client: ManifestClient = await ManifestClient.getClientForMarket(
+      connection,
+      marketAddress,
+      keypair,
+    );
+    return new DummyTrader(
+      connection,
+      keypair,
+      marketAddress,
+      name,
+      commands,
+      client,
+    );
+  }
+
+  public async run(): Promise<void> {
+    // Randomly take an action.
+    console.log('Running');
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      await new Promise((f) => setTimeout(f, 1000));
+      const market: Market = await Market.loadFromAddress({
+        connection: this.connection,
+        address: this.marketAddress,
+      });
+      market.prettyPrint();
+
+      const randomFunction =
+        this.commands[Math.floor(Math.random() * this.commands.length)];
+      switch (randomFunction) {
+        case 'DEPOSIT': {
+          console.log(`${this.name} trying to deposit`);
+          await this.deposit();
+          break;
+        }
+        case 'WITHDRAW': {
+          console.log(`${this.name} trying to withdraw`);
+          await this.withdraw();
+          break;
+        }
+        case 'PLACE_ORDER': {
+          console.log(`${this.name} trying to place order`);
+          await this.placeOrder();
+          //await this.placeOrder();
```

Comment:
extra comment

Reply:
intentional. similar to below with the functions enabling and disabling. want to quickly turn on and off stuff for testing without wiring up command line arguments. This one is to address how we cancel as much as we place, so there really never builds up an orderbook